app/src/main/java/org/schabi/newpipe/util/external_communication/ShareUtils.java

```diff
@@ -203,36 +127,82 @@ private static void openAppChooser(@NonNull final Context context,
+     *                   app or browser (the default app associated with this link or the default
+     *                   browser if there is no default app will be only shown in the system
+     *                   chooser on Android 12 and higher)
+     * @return whether the URL can be opened
+     */
+    public static boolean openUrlInBrowser(@NonNull final Context context,
+                                           final String url,
+                                           final boolean useChooser) {
+        final Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(url))
+                .setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+
+        if (useChooser) {
+            return openAppChooser(context, intent, context.getString(R.string.open_with));
+        }
+
+        return openIntentInApp(context, intent, true);
+    }
+
+    /**
+     * Open an intent with the system default app or a system chooser in the case multiple
+     * activities can handle the intent.
+     *
      * <p>
-     * If no app is set as default, it will return "android" (not on some devices because some
-     * OEMs changed the app chooser).
+     * The intent can be of every type, excepted a web intent for which
+     * {@link #openUrlInBrowser(Context, String, boolean)} should be used directly.
+     * </p>
+     *
      * <p>
-     * If no app is installed on user's device to handle the intent, it will return an empty string.
+     * If no app can open the intent, a toast with the message {@code No app on your device can
+     * open this} is shown if {@code showNoAppToOpenToast} is set to true.
+     * </p>
      *
-     * @param context the context to use
-     * @param intent  the intent to get default app
-     * @return the package name of the default app, an empty string if there's no app installed to
-     * handle the intent or the app chooser if there's no default
+     * @param context              the context to use
+     * @param intent               the intent to open
+     * @param showNoAppToOpenToast whether a toast is displayed to the user when no app can open
+     *                             the given intent
+     * @return whether the intent can be opened
      */
-    private static String getDefaultAppPackageName(@NonNull final Context context,
-                                                   @NonNull final Intent intent) {
-        final ResolveInfo resolveInfo = context.getPackageManager().resolveActivity(intent,
-                PackageManager.MATCH_DEFAULT_ONLY);
-
-        if (resolveInfo == null) {
-            return "";
-        } else {
-            return resolveInfo.activityInfo.packageName;
+    public static boolean openIntentInApp(@NonNull final Context context,
+                                          @NonNull final Intent intent,
+                                          final boolean showNoAppToOpenToast) {
```

Comment:
What about creating two methods and removing this `boolean` parameter? I would call the two methods:
- `tryOpenIntentInApp` which just does `try{...}catch{return false}; return true`
- `openIntentInApp` which does `if (!tryOpenIntentInApp()) showToast` and has return type `void`